use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+
+#### Calculating Team Score based on Event Time
+
+We want our pipeline to also output the total score for each team during each hour of play. Unlike the user score calculation, for team scores, we care about when in _event_ time each score actually occurred, because we want to consider each hour of play individually. We also want to provide speculative updates as each individual hour progresses, and to allow any instances of late data—data that arrives after a given hour's data is considered complete—to be included in our calculation.
+
+Because we consider each hour individually, we can apply fixed-time windowing to our input data, just like in `HourlyTeamScore`. To provide the speculative updates and updates on late data, we'll specify additional trigger parameters. The trigger will cause each window to calculate and emit results at an interval we specify (in this case, every five minutes), and also to keep triggering after the window is considered "complete" to account for late data. Just like the user score calculation, we'll set the trigger to accumulating mode to ensure that we get a running sum for each hour-long window.
+
+> **Note:** `LeaderBoard` reads its input data from Pub/Sub. We don't have to manually extract the timestamp from the data itself; the timestamp is derived from a Pub/Sub data element attribute.
+
+The triggers for speculative updates and late data help with the problem of [time skew](/learn/programming-guide/#windowing). Events in the pipeline aren't necessarily processed in the order in which they actually occurred according to their timestamps; they may arrive in the pipeline out of order, or late (in our case, because they were generated while the user's phone was out of contact with a network). Beam needs a way to determine when it can reasonably assume that it has "all" of the data in a given window: this is called the _watermark_.
+
+In an ideal world, all data would be processed immediately when it occurs, so the processing time would be equal to (or at least have a linear relationship to) the event time. However, because distributed systems contain some inherent inaccuracy (like our late-reporting phones), Beam often uses a heuristic watermark.
+
+The following diagram shows the relationship between ongoing processing time and each score's event time for two teams:
+
+<figure id="fig4">
+    <img src="/images/gaming-example-event-time-narrow.gif"
+         width="900" height="390">
+</figure>
+Figure 4: Score data by team, windowed by event time. A trigger based on processing time causes the window to emit speculative early results and include late results.
+
+The dotted line in the diagram is the "ideal" **watermark**: Beam's notion of when all data in a given window can reasonably considered to have arrived. The irregular solid line represents the actual watermark, as determined by the data source (in our case, Pub/Sub).
```

Comment:
Ah, here's the dashed line. I still it might be clearer to introduce it earlier?